### ISSUE TITLE
[no-unused-vars] False positive for class member constructor arguments

### ISSUE THREAD

**Issue by lgarron** [typescript-eslint/typescript-eslint#1312]
**Repro**

```ts
    export class MyClass {
      constructor(private X: string, private Y: string, public Z: string) {
        console.log(this.X);
      }
    }
```

Config is from `npx eslint --init`:
<details>
<pre>
module.exports = {
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended"
  ],
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly"
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "plugins": [
    "@typescript-eslint"
  ],
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "double"
    ],
    "semi": [
      "error",
      "always"
    ]
  }
};
</pre>
</details>

**Expected Result**

No errors.

**Actual Result**

    'X' is defined but never used. eslint(no-unused-vars)
    'Y' is defined but never used. eslint(no-unused-vars)
    'Z' is defined but never used. eslint(no-unused-vars)

All three are false positives, as far as I'm concerned:

- X is actually used in the code itself.
- Y is meant to be used in subclasses.
- Z is meant to be publicly accessible, and therefore doesn't have to be used directly in the class (or a subclass).

**Additional Info**

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.10.0` |
| `@typescript-eslint/parser`               | `2.10.0` |
| `TypeScript`                              | `3.5.2` |
| `ESLint`                                  | `6.7.2` |
| `node`                                    | `12.4.0` |
| `npm`                                     | `6.13.0` |

Possibly related:
https://github.com/typescript-eslint/typescript-eslint/issues/52
https://github.com/typescript-eslint/typescript-eslint/issues/586

I haven't ever had issues with this in TSLint, but I tried out ESLint because it's getting deprecated: https://github.com/palantir/tslint/issues/4534

I immediately get this false positive all over my project when I try to switch to ESLint, and this is unfortunately a dealbreaker for now. :-/

**Comment by bradzacher** [comment]
please use our `no-unused-vars-experimental` rule.
